src/PyListProxyHandler.cc

```diff
@@ -550,15 +550,7 @@ static bool array_concat(JSContext *cx, unsigned argc, JS::Value *vp) {
   PyObject *self = JS::GetMaybePtrFromReservedSlot<PyObject>(proxy, PyObjectSlot);
 
   Py_ssize_t selfSize = PyList_GET_SIZE(self);
-
-  PyObject *result = PyList_New(selfSize);
-
-  // Copy items to the new list
-  for (Py_ssize_t index = 0; index < selfSize; index++) {
-    PyObject *item = PyList_GetItem(self, index);
-    Py_INCREF(item); // `PyList_SetItem` steals the reference, so we must increase the reference count by 1
-    PyList_SetItem(result, index, item);
-  }
+  PyObject *result = PyList_GetSlice(self, 0, selfSize);
```

Comment:
Does this mean that python lists no longer change when JS arrays change?

Reply:
python lists proxied by their respective object change when the proxy object is changed.
However, this function implements Array.prototype.concat, which returns a new array, so this behaviour is correct.